microsoft/surface/default.nix

```diff
@@ -1,46 +1,12 @@
-{ lib, pkgs, ... }:
+{ lib, ... }:
 
 let
-  inherit (lib) mkDefault;
+  inherit (lib) warn;
 
 in {
   imports = [
-    ./kernel
-    ./firmware/surface-go/ath10k
+    ( warn
```

Comment:
There is a `warnings` nixos module option which might not break the test here. Otherwise we might need a workaround in our test script.

Reply:
> There is a `warnings` nixos module option which might not break the test here. Otherwise we might need a workaround in our test script.

Yeah, it wasn't clear whether this would be preferred over `lib.warn`.
I'll take a look.

Alternatively, I can perhaps use a simpler `builtins.trace` instead?

Reply:
> > There is a `warnings` nixos module option which might not break the test here. Otherwise we might need a workaround in our test script.
> 
> Yeah, it wasn't clear whether this would be preferred over `lib.warn`. I'll take a look.

I tested this, and it also triggers a fail, the same as using `lib.warn` directly.

> Alternatively, I can perhaps use a simpler `builtins.trace` instead?

If I want to provide a deprecation assertion or warning, it seems like this is my only option.
And it's not a really great option, realistically.

Reply:
> There is a `warnings` nixos module option which might not break the test here. Otherwise we might need a workaround in our test script.

Ok, to hack around this, I've kept the `//microsoft/surface/default.nix` file, but removed the entry from the `//README.md` file.
I've also converted the `warnings` to `assertions` instead.

WDYT?